decimal.go

```diff
@@ -126,7 +126,7 @@ func New(value int64, exp int32) Decimal {
 			}
 		}
 		// when exp > 7, it would be greater than maxInt
-		if exp <= 6 {
+		if exp >= 0 && exp <= 6 {
```

Comment:
I would suggest to either exclude zero or use `else if`. 
I think generally these two clauses have the same outcome when `exp == 0`
Hence, If I understood correctly, in case of exceeding the `minInt`/`maxInt` with `exp == 0` then flow will enter both clauses which is not needed. What do you think?
```suggestion
		if exp > 0 && exp <= 6 {
```